MetaMask/metamask-extension
# fix (cherry-pick): Prevent Token Autodetection Firefox blocker

## **Description**

Token detection modal seems to stay stuck on firefox

## Original PR

https://github.com/MetaMask/metamask-extension/pull/25279

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25279?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/25256

## **Manual testing steps**
To run locally use: 
`rm -rf dist && ENABLE_MV3=false yarn && ENABLE_MV3=false yarn build:dev dev --build-type main --lockdown false --snow false --apply-lavamoat false`

1. Import wallet on firefox
2. Got to settings => security and privacy and disable token auto detection
3. Go back to home page and you should see the token detection modal
4. Clicking on allow should dismiss the modal and enable the token detection


## **Screenshots/Recordings**



### **Before**




https://github.com/MetaMask/metamask-extension/assets/10994169/e23f59d6-7bc9-4707-872d-436c1a56e7cf


### **After**


https://github.com/MetaMask/metamask-extension/assets/10994169/cf47e04c-e5bc-4898-92f1-c80c87988fdb




## **Pre-merge author checklist**

Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).

## **Pre-merge reviewer checklist**


---------




## **Description**



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25309?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
